src/object/got-object.controller.ts

```diff
@@ -0,0 +1,49 @@
+import { Get, Controller, Param, Body, UseInterceptors, HttpException, HttpStatus, UsePipes, ValidationPipe, Post } from '@nestjs/common';
+import { GotObjectDto } from './dto/got-object.dto';
+import { TransformInterceptor } from '../common/interceptors/transform.interceptor';
+import uuid = require('uuid-v4');
+import { ApiResponse } from '@nestjs/swagger';
+import { GotObjectStorageService } from './got-object-storage.service';
+import { GotObjectValidationService } from './got-object-validation.service';
+
+@Controller('objects/object')
+@UseInterceptors(TransformInterceptor)
+export class GotObjectController {
+
+    /**
+     * @param  {GotObjectValidationService} privategotObjectValidationService
+     * @returns GotObjectValidationService
+     */
+    constructor(private gotObjectValidationService: GotObjectValidationService, 
+        private gotObjectStorageService: GotObjectStorageService) {
+    }
+
+    @ApiResponse({ status: 200, description: 'The record has been found.' })
+    @ApiResponse({ status: 404, description: 'The record has not been found.' })
+    @ApiResponse({ status: 400, description: 'Bad Request.' })
+    @Get('/:id')
+    public getObject(@Param() params: any): Promise<GotObjectDto> {
+        return this.gotObjectStorageService.get(params.id);
+    }
+
+    @ApiResponse({ status: 201, description: 'The record has been successfully created.' })
+    @ApiResponse({ status: 400, description: 'Bad Request.' })
+    @Post('/')
+    public storeObject(@Body() gotObject: GotObjectDto): Promise<any> {
```

Comment:
Wrong.

In REST API design you use either PUT or POST based on the indempotence of the action. 

Since the caller can create multiple objects that are similar on the same endpoint, without defining the object idea himself, POST is the only right way to go for creating objects and PUT would be the way to go for updating exisiting ones (to be implemented in another task).

See reference here :
https://stackoverflow.com/a/2691891

Reply:
*object ID not object idea